src/System.Runtime.Extensions/tests/System/IO/PathTestsBase.cs

```diff
@@ -201,235 +201,253 @@ public partial class PathTestsBase
+        public static TheoryData<string, string, string> GetFullPath_Windows_FullyQualified => new TheoryData<string, string, string>
         {
-            string curDir = @"/home/git";
-
-            return new TheoryData<string, string, string>
-            {
-                { curDir, curDir, curDir },
-                { "", curDir, curDir },
-                { "..", curDir, Path.GetDirectoryName(curDir) },
-                { Path.Combine(curDir, ".", ".", ".", ".", "."), curDir, curDir },
-                { curDir + new string(Path.DirectorySeparatorChar, 3) + ".", curDir, curDir },
-                { Path.Combine(curDir, "..", Path.GetFileName(curDir), ".", "..", Path.GetFileName(curDir)), curDir, curDir },
-                { Path.Combine(Path.GetPathRoot(curDir), "somedir", ".."), curDir, Path.GetPathRoot(curDir) },
-                { Path.Combine(Path.GetPathRoot(curDir), "."), curDir, Path.GetPathRoot(curDir) },
-                { Path.Combine(Path.GetPathRoot(curDir), "..", "..", "..", ".."), curDir, Path.GetPathRoot(curDir) },
-                { Path.GetPathRoot(curDir) + new string(Path.DirectorySeparatorChar, 3), curDir, Path.GetPathRoot(curDir) },
-                { "tmp", curDir, curDir + "/tmp" },
-                { "tmp/bar/..", curDir, curDir + "/tmp" },
-                { "tmp/..", curDir, curDir },
-                { "tmp/./bar/../", curDir, curDir + "/tmp/" },
-                { "tmp/bar/../../", curDir, curDir + "/" },
-                { "tmp/bar/../next/../", curDir, curDir + "/tmp/" },
-                { "tmp/bar/next", curDir, curDir + "/tmp/bar/next" },
+            // Device paths
+            { "foo", @"\\?\C:\ ", @"\\?\C:\ \foo" },
+            { @" \ \foo", @"\\?\C:\", @"\\?\C:\ \ \foo" },
+            { @" .\foo", @"\\?\C:\", @"\\?\C:\ .\foo" },
+            { @" ..\foo", @"\\?\C:\", @"\\?\C:\ ..\foo" },
+            { @"...\foo", @"\\?\C:\", @"\\?\C:\...\foo" },
+
+            { @"foo", @"\\?\GLOBALROOT\", @"\?\GLOBALROOT\foo" },
+            { @"foo", @"\\?\", @"\?\foo" },
+            { @".\foo", @"\\?\", @"\?\foo" },
+            { @"..\foo", @"\\?\", @"\foo" },
+            { @"foo", @"\\?\\", @"\?\foo" },
+
+            { @"foo", @"\\?\C:\\", @"\\?\C:\\foo" },
+            { @"|\foo", @"\\?\C:\", @"\\?\C:\|\foo" },
+            { @".\foo", @"\\?\C:\", @"\\?\C:\.\foo" },
+            { @"..\foo", @"\\?\C:\", @"\\?\C:\..\foo" },
+
+            { @"\Foo1\.\foo", @"\\?\C:\", @"\\?\C:\Foo1\foo" },
+            { @"\Foo2\..\foo", @"\\?\C:\", @"\\?\C:\foo" },
+
+            { "foo", @"\\.\C:\ ", @"\\.\C:\ \foo" },
+            { @" \ \foo", @"\\.\C:\", @"\\.\C:\ \ \foo" },
+            { @" .\foo", @"\\.\C:\", @"\\.\C:\ .\foo" },
+            { @" ..\foo", @"\\.\C:\", @"\\.\C:\ ..\foo" },
+            { @"...\foo", @"\\.\C:\", @"\\.\C:\...\foo" },
+
+            { @"foo", @"\\.\GLOBALROOT\", @"\GLOBALROOT\foo" },
```

Comment:
These `\\.\` paths are all bugs. Are you testing with the updated CoreCLR? I made a fix for GetRootPath for `\\.\`.  You should build coreclr yourself (with latest bits) to validate what you're getting here and comment out things that are broken without the new CLR with an issue to track.